### PR TITLE
gaussian prime

### LINKED ISSUE #18660: Implement gaussian prime
I see there are lots of ntheory stuff for gaussian integers, like finding gaussian prime
```python3
from sympy.ntheory.primetest import isprime
from sympy.core.compatibility import as_int

def is_gaussian_prime(a, b):
    """Test if a + b*I is a Gaussian prime number.
    
    References
    ==========
    
    .. [1] https://oeis.org/wiki/Gaussian_primes
    """
    a = as_int(a)
    b = as_int(b)
    if a == 0:
        b = abs(b)
        return isprime(b) and b % 4 == 3
    elif b == 0:
        a = abs(a)
        return isprime(a) and a % 4 == 3
    return isprime(a**2 + b**2) 

is_gaussian_prime(2, 5)
```

Similarly, I see that gaussian integers can have euclidean division, gcd, every stuff
Though it can be more complicated because the division may not be unique. 
But I wonder if any computational method can be able to generalize such finds.

At least using `ZZ[I]` seems like giving a wrong result for computing modulus.
```
from sympy import *

p = ZZ[I](3 + 2*I)
q = ZZ[I](-1 + 3*I)
p % q
```

### PR DESCRIPTION
**Pull request by abh2k**
Fixes #18660 

* ntheory
  * implemented gaussian prime test

### CODE REVIEW

File: sympy/ntheory/primetest.py [hunk 1]
```diff
@@ -643,3 +643,25 @@ def isprime(n):
...
+def is_gaussian_prime(num):
+    """Test if num is a Gaussian prime number.
+
+    References
+    ==========
+
+    .. [1] https://oeis.org/wiki/Gaussian_primes
+    """
+    from sympy import I
+    num += 0*I
```
Comment (sylee957) [inline sympy/ntheory/primetest.py]: Why is this needed?

Comment (abhinav28071999) [reply sympy/ntheory/primetest.py]: When num is integr, it shows `AttributeError: 'int' object has no attribute 'as_real_imag'`

Comment (sylee957) [reply sympy/ntheory/primetest.py]: I think that it's better to use sympify. There is no advantage of doing some sympify hacks like this, because addition already does sympify internally.

**Comment by sylee957** [comment]
And also, make sure that you check `rst` files in the docs, such that this is documented.

**Comment by sylee957** [comment]
You should address my last comment and add the doc under `doc/src/modules/ntheory.rst`.
It will be fine to merge after this.

**Comment by abhinav28071999** [comment]
> You should address my last comment and add the doc under `doc/src/modules/ntheory.rst`.
> It will be fine to merge after this.

Done.

**Comment by sylee957** [comment]
The sphinx error suggests that you should add under the scope of
.. module:: sympy.ntheory.primetest
